### DISCUSSION TITLE
How to create a service (AIServices) with memory and streaming responses?

### DISCUSSION THREAD

**Discussion by rgrin08** [langchain4j/langchain4j#801] [Q&A]
I would like to use a service with memory and streaming.
Code without streaming:
```
Assistant assistant = AiServices.builder(Assistant.class)
     .chatLanguageModel(model)
     .chatMemory(chatMemory)
     .build();
```
This code does not work if the type of the model is StreamingChatLanguageModel.
How can I do in this case?

**Comment by tenpigs267** [comment]
You should just use  .streamingChatLanguageModel instead of .chatLanguageModel
or use the constructor as in this example : https://github.com/langchain4j/langchain4j-examples/blob/main/other-examples/src/main/java/ServiceWithStreamingExample.java

**Reply by anunnakian** [reply]
Maybe this PR will help you https://github.com/langchain4j/langchain4j/pull/971
I merged the `ChatLanguageModel` and `StreamingChatLanguageModel` classes, so you will be able to do this : 

```
OpenAiChatModel.builder()
      .baseUrl(System.getenv("OPENAI_BASE_URL"))
      .apiKey(System.getenv("OPENAI_API_KEY"))
      .organizationId(System.getenv("OPENAI_ORGANIZATION_ID"))
      .logRequests(true)
      .logResponses(true)
      .isStreaming(true) // Enable streaming 
      .build()
```
